### DISCUSSION TITLE
Creating a scrollable frame in customtkinter

### DISCUSSION THREAD

**Discussion by S1lentAFK** [TomSchimansky/CustomTkinter#1066] [Q&A]
Hi, i have a question. How do I make my CTkframe scrollable? I tried setting the frame like this frame.yview, but it says thah the object has no attribute yview, So can somebody please help me? I really need help!!!

**Comment by FarhanRah** [comment]
Same question, interested in how to create a scrollable frame.

**Comment by VollGerne** [comment]
A frame is not scrollable directly. 
Create instead a tkinter.Canvas, then add a CTKFrame with 'window_configure' to that Canvas and make the Canvas scrollable.

**Comment by Kesatas** [comment]
im also interested how in to make CtkTabview tabview scrolable ...i have alot labels with entrys and it not fit on the tab_view... ;( can anyone help ?

> self.tabview = customtkinter.CTkTabview(self, width=750, height=500)
> self.tabview.grid(row=0, column=2, padx=(20, 0), pady=(2, 0), sticky="nsew")
> self.tabview.add("Collect")
> self.tabview.add("Talk")
> self.tabview.tab("Collect").grid_columnconfigure(1, weight=1)  # configure grid of individual tabs
> self.tabview.tab("Collect").grid_columnconfigure((2, 3), weight=0)
> self.tabview.tab("Collect").grid_rowconfigure((0, 1, 2), weight=0)
> self.scrollbar = customtkinter.CTkScrollbar(self.tabview.tab("Collect"), orientation="vertical")
> self.scrollbar.grid(row=0, column=3, rowspan=20, padx=(10, 10), pady=(10, 10), sticky="ns")
> alot labels, entrys........

**Comment by jh-jensen** [comment]
Iv'e done it like this (modified it myself from another solution from somewhere else):

```
import tkinter as tk
from tkinter import ttk
import customtkinter 

class CustomScrollableFrame(customtkinter.CTkFrame):
    def __init__(self, container, widthcanvas, heightcanvas, *args, **kwargs):
        super().__init__(container, widthcanvas, heightcanvas, *args, **kwargs)

        self.system_appearance_mode = customtkinter.get_appearance_mode()

        self.canvas_bg = "gray90" if self.system_appearance_mode == "Light" else "gray13"
        self.canvas = customtkinter.CTkCanvas(self, bg=self.canvas_bg, bd=0, highlightthickness=0, width=widthcanvas, height=heightcanvas)
        # canvas.configure(selectbackground=None, selectforeground=None)
        self.scrollbar = customtkinter.CTkScrollbar(self, orientation="vertical", command=self.canvas.yview)
        self.scrollable_frame = customtkinter.CTkFrame(self.canvas)
        # self.scrollable_frame = customtkinter.CTkFrame(self.canvas, fg_color="blue")

        self.scrollable_frame.bind(
            "<Configure>",
            lambda e: self.canvas.configure(
                scrollregion=self.canvas.bbox("all")
            ),
        )
        self.scrollable_frame.bind(
            "<MouseWheel>",
            lambda e: self.canvas.yview_scroll(int(-1*(e.delta/120)), "units"),
        )

        # Width might need adjusting
        self.canvas.create_window((0, 0), window=self.scrollable_frame, anchor="nw", width=widthcanvas+60)
        self.canvas.pack(side="left", fill="both", expand=True)
        self.canvas.configure(yscrollcommand=self.scrollbar.set)
        self.scrollbar.pack(side="right", fill="y")
```

**Reply by Kesatas** [reply]
thanks alot !
